cdap-app-fabric/src/main/java/io/cdap/cdap/support/handlers/SupportBundleHttpHandler.java

```diff
@@ -81,4 +127,205 @@ public void createSupportBundle(HttpRequest request, HttpResponder responder,
   }
+
+  /**
+   * Get the list of bundle status from status.json.
+   *
+   * @throws IOException is thrown if status.json is not found
+   */
+  @GET
+  @Path("/support/bundle")
+  public void getSupportBundle(HttpRequest request, HttpResponder responder) throws Exception {
+    /** ensure the user has authentication to create supportBundle */
+    contextAccessEnforcer.enforceOnParent(EntityType.SUPPORT_BUNDLE, InstanceId.SELF, StandardPermission.GET);
+    File baseDirectory = new File(cConf.get(Constants.CFG_LOCAL_DATA_SUPPORT_BUNDLE_DIR));
+    if (!baseDirectory.exists()) {
+      responder.sendString(HttpResponseStatus.OK, "No content in Support Bundle.");
+      return;
+    }
+    JsonArray statusJsonArray = new JsonArray();
+    File[] supportFiles =
+      baseDirectory.listFiles((dir, name) -> !name.startsWith(".") && !dir.isHidden() && dir.isDirectory());
+    if (supportFiles != null && supportFiles.length > 0) {
+      for (File uuidFile : supportFiles) {
+        JsonObject singleBundleJson = bundleService.getSingleBundleJson(uuidFile);
+        statusJsonArray.add(singleBundleJson);
+      }
+    }
+    responder.sendString(HttpResponseStatus.OK, GSON.toJson(statusJsonArray));
+  }
+
+  /**
+   * Get the specific bundle status from status.json by using uuid.
+   *
+   * @param uuid the bundle id which is also the uuid
+   * @throws IOException is thrown if status.json is not found
+   */
+  @GET
+  @Path("/support/bundle/{uuid}")
+  public void getSupportBundleByUUID(HttpRequest request, HttpResponder responder, @PathParam("uuid") String uuid)
+    throws Exception {
+    contextAccessEnforcer.enforceOnParent(EntityType.SUPPORT_BUNDLE, InstanceId.SELF, StandardPermission.GET);
+    File baseDirectory = new File(cConf.get(Constants.CFG_LOCAL_DATA_SUPPORT_BUNDLE_DIR));
+    File uuidFile = new File(cConf.get(Constants.CFG_LOCAL_DATA_SUPPORT_BUNDLE_DIR), uuid);
+    if (!baseDirectory.exists()) {
+      responder.sendString(HttpResponseStatus.OK, "No content in Support Bundle.");
+      return;
+    }
+    if (!uuidFile.exists()) {
+      responder.sendString(HttpResponseStatus.OK, String.format("No such uuid %s in Support Bundle.", uuid));
+      return;
+    }
+    JsonArray statusJsonArray = new JsonArray();
+    JsonObject singleBundleJson = bundleService.getSingleBundleJson(uuidFile);
+    statusJsonArray.add(singleBundleJson);
+    responder.sendString(HttpResponseStatus.OK, GSON.toJson(statusJsonArray));
+  }
+
+  /**
+   * Delete the specific bundle by using the uuid
+   *
+   * @param uuid the bundle id which is also the uuid
+   * @throws IOException is thrown if status.json is not found
+   */
+  @DELETE
+  @Path("/support/bundle/{uuid}")
+  public void deleteSupportBundle(HttpRequest request, HttpResponder responder, @PathParam("uuid") String uuid)
+    throws Exception {
+    contextAccessEnforcer.enforceOnParent(EntityType.SUPPORT_BUNDLE, InstanceId.SELF, StandardPermission.DELETE);
```

Comment:
This should probably use enforce instead of enforceOnParent and enforce upon the specific SupportBundleId instead of on the parent.